homework/unique_ptr/UniquePointer.hpp

```diff
@@ -140,30 +156,32 @@ class UniquePointer<T[]> {
     T& operator*();
     T* get() const;
     T* release();
-    void reset();
-    void reset(T* pointer);
+    void reset(std::function<void(T*)> deleter = defaultBlockDeleter<T>);
+    void reset(T* pointer, std::function<void(T*)> deleter = defaultBlockDeleter<T>);
```

Comment:
no need to have both functions. Do sth like this:
```suggestion
    void reset(T* pointer = nullptr, std::function<void(T*)> deleter = defaultBlockDeleter<T>);
```